cluster/src/main/java/com/hartwig/pipeline/tertiary/cuppa/Cuppa.java

```diff
@@ -0,0 +1,158 @@
+    private final InputDownload purpleQc;
+    private final InputDownload purplePurity;
+    private final InputDownload linxDriverCatalog;
+    private final InputDownload linxClusters;
+    private final InputDownload linxFusions;
+    private final InputDownload linxViralInsertions;
+    private final ResourceFiles resourceFiles;
+    private final PersistedDataset persistedDataset;
+
+    public Cuppa(final PurpleOutput purpleOutput, final LinxOutput linxOutput, final ResourceFiles resourceFiles,
+            final PersistedDataset persistedDataset) {
+        purpleSomaticVcfDownload = new InputDownload(purpleOutput.outputLocations().somaticVcf());
+        purpleStructuralVcfDownload = new InputDownload(purpleOutput.outputLocations().structuralVcf());
+        purpleQc = new InputDownload(purpleOutput.outputLocations().qcFile());
+        purplePurity = new InputDownload(purpleOutput.outputLocations().purityTsv());
+        linxDriverCatalog = new InputDownload(linxOutput.linxOutputLocations().driverCatalog());
+        linxClusters = new InputDownload(linxOutput.linxOutputLocations().breakends());
+        linxFusions = new InputDownload(linxOutput.linxOutputLocations().fusions());
+        linxViralInsertions = new InputDownload(linxOutput.linxOutputLocations().viralInsertions());
+        this.resourceFiles = resourceFiles;
+        this.persistedDataset = persistedDataset;
+    }
+
+    @Override
+    public List<BashCommand> inputs() {
+        return List.of(purpleSomaticVcfDownload,
+                purpleStructuralVcfDownload,
+                purpleQc,
+                purplePurity,
+                linxDriverCatalog,
+                linxClusters,
+                linxFusions,
+                linxViralInsertions);
+    }
+
+    @Override
+    public String namespace() {
+        return NAMESPACE;
+    }
+
+    @Override
+    public List<BashCommand> commands(final SomaticRunMetadata metadata) {
+        return List.of(new JavaJarCommand("cuppa",
+                        Versions.CUPPA,
+                        "cuppa.jar",
+                        "4G",
+                        List.of("-categories",
+                                "DNA",
+                                "-ref_data_dir",
+                                resourceFiles.cuppaRefData(),
+                                "-sample_data",
+                                metadata.tumor().sampleName(),
+                                "-sample_data_dir",
+                                VmDirectories.INPUT,
+                                "-sample_sv_file",
+                                purpleStructuralVcfDownload.getLocalTargetPath(),
+                                "-sample_somatic_vcf",
+                                purpleSomaticVcfDownload.getLocalTargetPath(),
+                                "-log_debug",
+                                "-output_dir",
+                                VmDirectories.OUTPUT)),
+                () -> join(" ",
```

Comment:
I think we should introduce a PythonCommand abstraction over using an anonymous lambda. Also looks strange to have the join rather than just making it a single list with two commands.